iceoryx2-bb/posix/src/deadline_queue.rs

```diff
@@ -265,10 +265,28 @@ impl DeadlineQueue {
         for attachment in &*self.attachments.borrow() {
             let duration_until_last = last.max(attachment.start_time) - attachment.start_time;
             let duration_until_now = now - attachment.start_time;
-            if (duration_until_last / attachment.period) < (duration_until_now / attachment.period)
-                && call(DeadlineQueueIndex(attachment.index)) == CallbackProgression::Stop
-            {
-                return;
+            match attachment.period {
+                0 => {
+                    if matches!(
```

Comment:
It is just my personal opinion but I would prefer `call(DeadlineQueueIndex(attachment.index)) == CallbackProgression::Stop` over `matches!(call(DeadlineQueueIndex(attachment.index)), CallbackProgression::Stop)` just because it is for me just more intuitive.

And also my internal pattern matching in my head has much less to think about a line like `if whatever == foo` than `if matches!(whatever, foo)`. Especially, when we have to deal with multiple languages like c, c++, rust and soon python or go.

So, I would prefer it the old way but I do not insist.